### PR TITLE
SONARJAVA-2483 enqueue all possible catch blocks

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ExplodedGraphWalker.java#L39](https://github.com/SonarSource/sonar-java/blob/42ec6c9162ef6021f73e5d0d3d7324af4a00f78f/java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java#L39): Remove this unused import 'java.util.Optional'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFG.java [hunk 1]
```diff
@@ -69,6 +72,10 @@ public String getExceptionType() {
       return exceptionType;
     }
 
+    public Type getExceptionType(SemanticModel semanticModel) {
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFG.java]: Maybe we can remove the `getExceptionType()` method then and only rely on this one. But I didn't tried.

File: its/ruling/src/test/resources/jdk6/squid-S2583.json [hunk 2]
```diff
@@ -28,9 +28,6 @@
 'jdk6:java/beans/ReflectionUtils.java':[
 217,
 ],
-'jdk6:java/net/SocketInputStream.java':[
-143,
```
Comment (m-g-sonar) [inline its/ruling/src/test/resources/jdk6/squid-S2583.json]: :+1:

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 3]
```diff
@@ -807,27 +807,23 @@ private void enqueueUncheckedExceptions() {
...
 
   private void enqueueExceptionHandlers(Type exceptionType, ProgramState ps) {
     programPosition.block.successors().stream()
       .map(b -> (BytecodeCFG.Block) b)
-      .filter(b -> b.getExceptionType() != null)
+      .filter(BytecodeCFG.Block::isCatchBlock)
       .filter(b -> b.isUncaughtException()
         ||/*required as long as there is no real type tracking*/ exceptionType == null
-        || exceptionType.isSubtypeOf(b.getExceptionType()))
+        || exceptionType.isSubtypeOf(b.getExceptionType(semanticModel))
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: I would try to spare two consecutive calls to `b.getExceptionType(semanticModel)`. Dedicated small method?

**Pull request merged**